edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/DiskSpaceCheckerBase.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage.RocksDb.Disk
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public abstract class DiskSpaceCheckerBase : IDisposable
+    {
+        readonly PeriodicTask storageSpaceChecker;
+        readonly object updateLock = new object();
+
+        DiskSpaceStatus diskStatus;
+
+        protected DiskSpaceCheckerBase(TimeSpan checkFrequency, ILogger logger)
+        {
+            this.Logger = Preconditions.CheckNotNull(logger);
+
+            // Start after 5secs to allow initialization to finish
+            this.storageSpaceChecker = new PeriodicTask(this.PeriodicTaskCallback, checkFrequency, TimeSpan.FromSeconds(5), logger, "Disk space check");
+        }
+
+        public DiskSpaceStatus DiskStatus
+        {
+            get
+            {
+                // If disk status is Critical / Full, check disk status every time
+                if (this.diskStatus > DiskSpaceStatus.Available)
+                {
+                    this.UpdateCurrentDiskSpaceStatus();
+                }
+
+                return this.diskStatus;
+            }
+        }
+
+        public void Dispose()
+        {
+            lock (this.updateLock)
+            {
+                this.storageSpaceChecker?.Dispose();
+            }
```

Comment:
I don't think you need the locking here. You're already locking in the caller DiskSpaceChecker which should suffice. I think it's okay to make this non thread-safe and let the caller handle the thread-safety.